signers/signer-evm/src/signer.ts

```diff
@@ -210,12 +209,9 @@ export class DefaultEvmSigner implements GenericSigner<EvmTransaction> {
       return { hash: txHash };
     } catch (err) {
       const error = err as any; // TODO find a proper type
-      if (
-        error?.code === RPCErrorCode.TRANSACTION_REPLACED &&
-        error?.replacement
-      ) {
-        return { hash: error?.replacement?.hash, response: error?.replacement };
-      } else if (error?.code === RPCErrorCode.CALL_EXCEPTION) {
+      if (isError(error, 'TRANSACTION_REPLACED')) {
+        return { hash: error.replacement.hash, response: error.replacement };
+      } else if (isError(error, 'CALL_EXCEPTION')) {
```

Comment:
Please also handle canceled scenario like this.
```ts
      if (isError(error, "TRANSACTION_REPLACED")) {
        const reason = error.reason
        if (reason === "cancelled") {
          throw new SignerError(
            SignerErrorCode.SEND_TX_ERROR,
            undefined,
            'Transaction replaced and canceled by user',
            undefined,
            error
          );
        }
        return { hash: error.replacement.hash, response: error.replacement };
      }
```

Reply:
Tnx fixed.